client/lib/menu-data/menu-data.js

```diff
@@ -878,6 +878,11 @@ MenuData.prototype.fetchDefaultMenu = function() {
 			return;
 		}
 
+		if ( data.__syncResponse ) {
+			debug( 'ignoring second response here ...' );
+			return;
+		}
```

Comment:
I understand you're doing this to avoid an issue with the double-callback approach in menus, but I think we should instead just re-work menus to handle the double-callback. I'll take a look at that.